dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/Symbol.java

```diff
@@ -47,21 +46,10 @@ String getClassBinaryName() {
   }
 
   @Override
-  public boolean equals(Object other) {
-    if (this == other) {
-      return true;
-    }
-    if (other == null || getClass() != other.getClass()) {
-      return false;
-    }
```

Comment:
> two instances of different subclasses could theoretically compare equal.

Doesn't this if statement prevent that?

Reply:
You're right. It does.

It still means subclasses can be created with additional data that is not considered in an equality comparison. Mostly it's a little disconcerting to have an equals method that should always be overridden, especially in an abstract class.

Reply:
> It still means subclasses can be created with additional data that is not considered in an equality comparison. 

I'm not sure this change is an improvement on that. Even with this PR, a subclass can declare an additional data that's not considered in equality comparison.